File: HackerRank/Fizz Buzz.cpp
/*
Q: Fizz Buzz
   
   short program that prints each number from 1 to 100 on a new line. 

   > For each multiple of 3, print "Fizz" instead of the number. 

   > For each multiple of 5, print "Buzz" instead of the number. 

   > For numbers which are multiples of both 3 and 5, print "FizzBuzz" instead of the number.
   
   example output: 
   
   1, 2, Fizz, 4, Buzz, Fizz, 7, 8, Fizz, Buzz, 11, Fizz, 13, 14, 
   Fizz Buzz, 16,....etc

Notes:
------

Simple login using if and else if, the only catch is that you will get error if you implement as per the order of the statement. The logic
has to be implemented from the bottom most statement as it involves two conditions. 

Output screenshot: https://github.com/Ranjul-Arumadi/Coding-Problems/blob/main/Output%20Screenshots/Fizz%20Buzz.jpg
*/

/*-------------------------Solution in C++-------------------------*/
#include <iostream>
using namespace std;
int main()
{
    for(int i=1;i<=100;i++){
        if((i%3 == 0) && (i%5==0))
            cout<<"FizzBuzz\n";
        else if(i%3 == 0)
            cout<<"Fizz\n";
        else if(i%5 == 0)
            cout<<"Buzz\n";
        else
            cout<<i<<"\n";
     }
    return 0;
}

File: LeetCode/Algorithms/Easy/FizzBuzz.java
/*
    link: https://leetcode.com/problems/fizz-buzz/
    Given an integer n, return a string array answer (1-indexed) where:

    answer[i] == "FizzBuzz" if i is divisible by 3 and 5.
    answer[i] == "Fizz" if i is divisible by 3.
    answer[i] == "Buzz" if i is divisible by 5.
    answer[i] == i (as a string) if none of the above conditions are true.


    Example 1:

    Input: n = 3
    Output: ["1","2","Fizz"]
*/
class Solution {
    public List<String> fizzBuzz(int n) {
        List<String> l = new ArrayList<String>();
       
        for(int i=1;i<=n;i++){
            if(i%3==0 && i%5==0){
                l.add("FizzBuzz");
            }
            else if(i%3==0){
                l.add("Fizz");
            }
            else if(i%5==0){
                l.add("Buzz");
            }
            else{
                l.add(Integer.toString(i));
            }
        }
       
        return l;
    }
}
